src/dune_rules/odoc.ml

```diff
@@ -791,18 +808,31 @@ let setup_lib_html_rules_def =
     f
 ;;
 
-let setup_lib_html_rules sctx lib =
-  Memo.With_implicit_output.exec setup_lib_html_rules_def (sctx, lib)
+let setup_lib_html_rules sctx ~search_db lib =
+  Memo.With_implicit_output.exec (setup_lib_html_rules_def ~search_db) (sctx, lib)
 ;;
 
 let setup_pkg_html_rules_def =
   let f (sctx, pkg) =
     let ctx = Super_context.context sctx in
+    let target = Pkg pkg in
     let* libs = libs_of_pkg ctx ~pkg in
-    let* () = Memo.parallel_iter libs ~f:(setup_lib_html_rules sctx)
+    let dir = Paths.html ctx target in
+    let* search_db =
+      let odocl_file artefact = artefact.odocl_file in
+      let* package_odocl = target |> odoc_artefacts sctx >>| List.map ~f:odocl_file in
+      let* lib_odocls =
+        libs
+        |> Memo.List.concat_map ~f:(fun lib -> odoc_artefacts sctx (Lib lib))
```

Comment:
all of that is already computed in the non-sherlodoc paths. I'll push something to factor this

Reply:
done